docs/enhancements/tenancy-based-auth.md

```diff
@@ -0,0 +1,83 @@
+To align with the ACM authorization scheme, we would like to allow the following:
+- A tenancy-based read-only access to resources. 
+- Users should be allowed to grant editing capabilities to their resources to other users in their organization.
+
+## Goals
+The following tables details the access permission rules for various resources:
+
+For clusters and cluster-based resources (all APIs that starts with `/v2/clusters/{cluster_id}`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List Cluster   | Returns clusters owned by all members of the organization                                      |
+| Get  Cluster   | Allowed for all members of the organization                                                    |
+| Edit Cluster   | Allowed for the owner and for each user that has update access for this cluster's subscription |
+| Delete Cluster | Allowed for the owner and for each user that has remove access for this cluster's subscription |
+
+For InfraEnv and InfraEnv-based resources like hosts (all APIs that starts with `/v2/infra-envs/{infra_env_id`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List InfraEnvs | For bound InfraEnvs, returns resources owned by all members of the organization<br>If unbound InfraEnvs, return resources if owned by the caller                                                                |
+| Get    InfraEnv| For bound InfraEnv, allowed for all members of the organization<br>For unbound InfraEnv allowed for the owner                                                                                                         |
+| Edit InfraEnv  | For bound InfraEnv, allowed for the owner and for each user that has update access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+| Delete InfraEnv| For bound InfraEnv, allowed for the owner and for each user that has remove access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+
+For day2 cluster operations:
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| Add Host       | Allowed for every user who has update access for the day1's cluster subscription               |
+
+Filtering for you own clusters:
+Get Cluster call shall have an additional query parameter which filters the cluster by owners only:
+`GET /v2/clusters?owneronly=true`
```

Comment:
are we aligned with the UI / UX on such functionality? 
@jelkosz

Reply:
Shouldn't the UI read the organization level List? In this way we would be aligned with OCM view with no effort.

Reply:
The AI and OCM cluster should have the exact same permissions since the UI will not be treating them separately but as two portions on the same entity. If this will be the case, the UI wont need any modifications. Will it be the case?

@jkilzi FYI